src/components/ProfileEditModal/ProfileEditModal.tsx

```diff
@@ -1,33 +1,79 @@
-import React, { useState } from "react";
+import React, { useState, useContext, useEffect } from "react";
 import styles from "./styles";
 import { Modal, View, Pressable, TextInput, Text } from "react-native";
 import SvgIcon from "../../../assets/SvgIcon";
 import globalStyles from "../../globalStyles";
+import { User } from "../../types/schema";
+import userContext from "../../context/userContext";
+import firebaseApp from "../../firebase/firebaseApp";
+import { getUser } from "../../firebase/firestore/user";
+
+export type UserUpdateData = {
+  full_name?: string;
+  role?: string;
+};
+
+type ProfileEditModalProps = {
+  visible: boolean;
+  setVisible: Function;
+  user: User;
+  setUser: any;
+};
+
+export const ProfileEditModal = ({
+  visible,
+  setVisible,
+  user,
+  setUser,
+}: ProfileEditModalProps) => {
+  const db = firebaseApp.firestore().collection("users");
+  const value = useContext(userContext);
 
-export const ProfileEditModal = ({ visible, setVisible, user }: any) => {
   const [nameText, onChangeNameText] = useState(user.full_name);
   const [roleText, onChangeRoleText] = useState(user.role);
 
+  const [newUserData, setNewUserData] = useState<UserUpdateData>({});
+  const [wasEdited, setWasEdited] = useState(false);
+
+  const refresh = () => {};
+
+  useEffect(() => {
+    getUser(value).then((currUser) => {
+      setUser(currUser);
+      onChangeNameText(user.full_name);
+      onChangeRoleText(user.role);
+    });
+  }, [visible]);
+
+  const handleSubmit = async (): Promise<void> => {
+    try {
+      db.doc(user.user_id).update({
```

Comment:
plz move any firebase content into our firebase folder
So this would be good under user.ts